tools/sz_repo_cli/lib/src/commands/src/deploy_ios_command.dart

```diff
@@ -0,0 +1,254 @@
+      )
+      ..addOption(
+        whatsNewOptionName,
+        help:
+            "Release notes either for TestFlight or App Store review submission. Describe what's new in this version of your app, such as new features, improvements, and bug fixes. The string should not exceed 4000 characters. Example usage: --whats-new 'Bug fixes and performance improvements.'",
+      )
+      ..addOption(
+        flavorOptionName,
+        allowed: _iosFlavors,
+        help: 'The flavor to build for. Only the "prod" flavor is supported.',
+        defaultsTo: 'prod',
+      );
+  }
+
+  static const privateKeyOptionName = 'private-key';
+  static const keyIdOptionName = 'key-id';
+  static const issuerIdOptionName = 'issuer-id';
+  static const releaseStageOptionName = 'stage';
+  static const flavorOptionName = 'flavor';
+  static const whatsNewOptionName = 'whats-new';
+
+  @override
+  String get description =>
+      'Deploys the Sharezone iOS app to the App Store or TestFlight. Automatically bumps the build number and builds the app. Codemagic CLI tools are required.';
+
+  @override
+  String get name => 'ios';
+
+  @override
+  Future<void> run() async {
+    _throwIfFlavorIsNotSupportForDeployment();
+
+    // Is used so that runProcess commands print the command that was run. Right
+    // now this can't be done via an argument.
+    //
+    // This workaround should be addressed in the future.
+    isVerbose = true;
+
+    final buildNumber = await _getNextBuildNumber();
+    await _buildApp(buildNumber: buildNumber);
+    await _publish();
+
+    print('Deployment finished 🎉 ');
+  }
+
+  void _throwIfFlavorIsNotSupportForDeployment() {
+    final flavor = argResults![flavorOptionName] as String;
+    if (flavor != 'prod') {
+      throw Exception(
+        'Only the "prod" flavor is supported for iOS deployment.',
+      );
+    }
+  }
+
+  Future<void> _publish() async {
+    final whatsNew = argResults![whatsNewOptionName] as String?;
+    final stage = argResults![releaseStageOptionName] as String;
+    final issuerId = argResults![issuerIdOptionName] as String? ??
+        Platform.environment['APP_STORE_CONNECT_ISSUER_ID'];
+    final keyIdentifier = argResults![keyIdOptionName] as String? ??
+        Platform.environment['APP_STORE_CONNECT_KEY_IDENTIFIER'];
+    final privateKey = argResults![privateKeyOptionName] as String? ??
+        Platform.environment['APP_STORE_CONNECT_PRIVATE_KEY'];
+
+    final isStable = stage == 'stable';
+    await runProcess(
+      'app-store-connect',
+      [
+        'publish',
+        '--path',
+        'build/ios/ipa/*.ipa',
+        '--release-type',
+        'AFTER_APPROVAL',
```

Comment:
```suggestion
        // The app version will be automatically released right after
        // it has been approved by App Review.
        '--release-type',
        'AFTER_APPROVAL',
```

I e.g. thought that this would mean that the version is only published after **we** approved it manually in the app store connect website.

(Note: Comment might be wrongly formatted)